rtgui/compressgamut.cc

```diff
@@ -0,0 +1,366 @@
+   
+    colorspace = Gtk::manage(new MyComboBoxText());
+    colorspace->append(M("TP_COMPRESSGAMUT_REC2020"));
+    colorspace->append(M("TP_COMPRESSGAMUT_PROPHOTO"));
+    colorspace->append(M("TP_COMPRESSGAMUT_ADOBE"));
+    colorspace->append(M("TP_COMPRESSGAMUT_SRGB"));
+    colorspace->append(M("TP_COMPRESSGAMUT_DCIP3"));
+    colorspace->append(M("TP_COMPRESSGAMUT_ACESP1"));
+    colorspace->set_active(3);
+    iVBox->pack_start(*colorspace);
+    iFrame->add(*iVBox);
+    pack_start(*iFrame);
+    colorspaceconn = colorspace->signal_changed().connect(sigc::mem_fun(*this, &Compressgamut::colorspaceChanged));
+
+    // Percentage of the core gamut to protect Limits
+    // Values calculated to protect all the colors of the ColorChecker Classic 24 as given by
+    // ISO 17321-1 and Ohta (1997)
+
+    th_c = Gtk::manage (new Adjuster (M("TP_COMPRESSGAMUT_CYANTH"), 0., 0.999, 0.001, 0.815));//0.999 to avoid: 1 - th = 0
+    th_m = Gtk::manage (new Adjuster (M("TP_COMPRESSGAMUT_MAGENTATH"), 0., 0.999, 0.001, 0.803));
+    th_y = Gtk::manage (new Adjuster (M("TP_COMPRESSGAMUT_YELLOWTH"), 0., 0.999, 0.001, 0.880));
+
+    Gtk::Frame *thFrame = Gtk::manage(new Gtk::Frame(M("TP_COMPRESSGAMUT_THRESHOLD")));
+    thFrame->set_label_align(0.025, 0.5);
+    Gtk::Box *thVBox = Gtk::manage(new Gtk::Box(Gtk::ORIENTATION_VERTICAL));
+    thFrame->set_tooltip_markup (M("TP_COMPRESSGAMUT_THRESHOLD_TOOLTIP"));
+
+    thVBox->pack_start (*th_c);
+    thVBox->pack_start (*th_m);
+    thVBox->pack_start (*th_y);
+    thFrame->add(*thVBox);
+    pack_start(*thFrame, Gtk::PACK_SHRINK);
+    // from ACES https://docs.acescentral.com/specifications/rgc/#appendix-c-illustrations
+    // https://docs.acescentral.com/specifications/rgc/#appendix-d-ctl-reference-implementation
+    // Distance from achromatic which will be compressed to the gamut boundary
+    // Values calculated to encompass the encoding gamuts of common digital cinema cameras
+    d_c = Gtk::manage (new Adjuster (M("TP_COMPRESSGAMUT_CYANLIM"), 1.001, 2.0, 0.001, 1.147));
+    d_m = Gtk::manage (new Adjuster (M("TP_COMPRESSGAMUT_MAGENTALIM"), 1.001, 2.0, 0.001, 1.264));
+    d_y = Gtk::manage (new Adjuster (M("TP_COMPRESSGAMUT_YELLOWLIM"), 1.001, 2.0, 0.001, 1.312));
+
+    Gtk::Frame *limFrame = Gtk::manage(new Gtk::Frame(M("TP_COMPRESSGAMUT_LIMIT")));
+    limFrame->set_label_align(0.025, 0.5);
+    limFrame->set_tooltip_markup (M("TP_COMPRESSGAMUT_LIMIT_TOOLTIP"));
+
+    Gtk::Box *limVBox = Gtk::manage(new Gtk::Box(Gtk::ORIENTATION_VERTICAL));
+    limVBox->pack_start (*d_c);
+    limVBox->pack_start (*d_m);
+    limVBox->pack_start (*d_y);
+    limFrame->add(*limVBox);
+    pack_start(*limFrame, Gtk::PACK_SHRINK);
+
+    // Aggressiveness of the compression curve Pwr
+    rolloff = Gtk::manage(new Gtk::CheckButton(M("TP_COMPRESSGAMUT_ROLLOFF")));
+    pwr = Gtk::manage (new Adjuster (M("TP_COMPRESSGAMUT_PWR"), 0.5, 2.0, 0.01, 1.2));
```

Comment:
According to [this](https://paper.dropbox.com/doc/ACES-Gamut-Mapping-Architecture-VWG-Technical-Documentation-Deliverable-tZHiuOCj0RdYw8PPkrTam), the power "needs to remain above one, but cannot go much higher than 1.5". I'm not sure why the lower limit is 1 because the curve is well behaved down to 0.

They do mention that 1.1 or 1.2 are good for SDR output and 1.4 is good for HDR. It is a very small range, so maybe we should keep the range small here too?

Reply:
From the tests I was able to do, on values ​​lower than 1, I took 0.5 and it works correctly